x/dex/keeper/tick_liquidity.go

```diff
@@ -7,15 +7,52 @@ import (
 )
 
 // SetTickLiquidity set a specific tickLiquidity in the store from its index
-func (k Keeper) SetTickLiquidity(ctx sdk.Context, tickLiquidity types.TickLiquidity) {
+func (k Keeper) SetTickLiquidityPoolReserves(ctx sdk.Context, pool types.PoolReserves) {
+	//Wrap pool back into TickLiquidity
+	tick := types.TickLiquidity{
+		Liquidity: &types.TickLiquidity_PoolReserves{
+			PoolReserves: &pool,
+		},
+	}
+
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.TickLiquidityKeyPrefix))
+	b := k.cdc.MustMarshal(&tick)
+	store.Set(types.TickLiquidityKey(
+		pool.PairId,
+		pool.TokenIn,
+		pool.TickIndex,
+		types.LiquidityTypePoolReserves,
+		pool.Fee,
+	), b)
+}
+
+func (k Keeper) SetTickLiquidityLimitOrder(ctx sdk.Context, tranche types.LimitOrderTranche) {
```

Comment:
IMO we should not specialize these KV getters and setters to the underlying dynamics of the TickLiquidity object. Better to have another function so that we can do:
```
k.SetTickLiquidity(NewTickLiquidity(tranche))
```

Reply:
See comment on polymorphism above